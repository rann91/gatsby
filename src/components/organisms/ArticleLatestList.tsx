/* @jsx jsx */
import { graphql, useStaticQuery } from 'gatsby'
import { Box, Container, Flex, jsx } from 'theme-ui'
import { column } from '../../gatsby-plugin-theme-ui'
import {
  Strapi_Article,
  Strapi_ComponentContentLatestArticleList
} from '../../typings/graphql'
import Section from '../atoms/Section'
import ArticleTeaser from '../molecules/ArticleTeaser'
import HeadingGroup from '../molecules/HeadingGroup'

interface Data {
  strapi: {
    articles: Strapi_Article[]
  }
}

const ArticleLatestList = ({
  title,
  subtitle,
  limit
}: Strapi_ComponentContentLatestArticleList) => {
  const data = useStaticQuery<Data>(graphql`
    query {
      strapi {
        articles(sort: "createdAt:desc") {
          ...ArticleTeaser
        }
      }
    }
  `)

  return (
    <Section data-testid="article-latest-list" hasTitle={!!title}>
      <Container>
        <HeadingGroup {...{ title, subtitle }} />
        <Flex
          sx={{
            flexWrap: 'wrap',
            '* + &': {
              paddingTop: [0, null, null, 5]
            }
          }}>
          {data.strapi.articles.slice(0, limit || 3).map(article => (
            <Box
              key={article.id}
              p={5}
              sx={{
                width: [column(12), null, null, column(4)],
                textAlign: ['center', null, null, 'left']
              }}>
              <ArticleTeaser {...article} />
            </Box>
          ))}
        </Flex>
      </Container>
    </Section>
  )
}

export default ArticleLatestList
