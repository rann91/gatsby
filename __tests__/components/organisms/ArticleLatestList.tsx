import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'
import React from 'react'
import ArticleLatestList from '../../../src/components/organisms/ArticleLatestList'
import data from '../../fixtures/article-list-latest.json'
import component from '../../fixtures/component-latest-articles.json'

beforeEach(() => {
  ;(useStaticQuery as jest.Mock).mockImplementationOnce(() => data)
})

describe('ArticleLatestList', () => {
  it('renders correctly', () => {
    const { container } = render(<ArticleLatestList {...component} />)

    expect(container).toMatchSnapshot()
  })
})
