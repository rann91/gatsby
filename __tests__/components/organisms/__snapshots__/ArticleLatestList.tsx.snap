// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleLatestList renders correctly 1`] = `
<div>
  <section
    class="css-1ptauik"
    data-testid="article-latest-list"
  >
    <div
      class="css-1yrmf69"
    >
      <div
        class="css-1mpaekt-HeadingGroup"
      >
        <div
          class="css-146vepc-HeadingGroup"
          data-testid="subtitle"
        >
          About us
        </div>
        <h2
          class="css-16nj6cn-HeadingGroup"
          data-testid="title"
        >
          Company news
        </h2>
      </div>
      <div
        class="css-18ozqqn-ArticleLatestList"
      >
        <div
          class="css-qzbx0h"
        >
          <h3
            class="css-9ygwq5-ArticleTeaser"
            data-testid="article-teaser-heading"
          >
            <a
              class="css-7aa45z-ArticleTeaser"
              href="/blog/10-quick-tips-about-blogging"
            >
              10 Quick Tips About Blogging
            </a>
          </h3>
          <time
            class="css-1muxnm4-ArticleTeaser"
            data-testid="article-teaser-label"
          >
            Jul 25, 2020
          </time>
          <div
            class="css-1muxnm4-ArticleTeaser"
            data-testid="article-teaser-summary"
          >
            Sit quam quae perferendis harum amet dolorum possimus minus. Occaecati magnam animi quasi dolor quae et aut. Autem voluptas alias nobis in odit. Aperiam harum eum modi neque. Ipsum iusto beatae et esse nobis illum hic natus. Sapiente non cumque quia facilis. Quo est odio culpa eaque sit ipsum mol.
          </div>
        </div>
        <div
          class="css-qzbx0h"
        >
          <h3
            class="css-9ygwq5-ArticleTeaser"
            data-testid="article-teaser-heading"
          >
            <a
              class="css-7aa45z-ArticleTeaser"
              href="/blog/15-best-blogs-to-follow-about-web-design"
            >
              15 Best Blogs To Follow About Web Design
            </a>
          </h3>
          <time
            class="css-1muxnm4-ArticleTeaser"
            data-testid="article-teaser-label"
          >
            Jul 25, 2020
          </time>
          <div
            class="css-1muxnm4-ArticleTeaser"
            data-testid="article-teaser-summary"
          >
            Maxime repellendus beatae asperiores. Est sapiente a et consectetur quo voluptate explicabo quis. Dignissimos error esse sapiente aperiam amet eum perspiciatis. Dolor totam voluptatum veritatis ex assumenda ut assumenda accus.
          </div>
        </div>
        <div
          class="css-qzbx0h"
        >
          <h3
            class="css-9ygwq5-ArticleTeaser"
            data-testid="article-teaser-heading"
          >
            <a
              class="css-7aa45z-ArticleTeaser"
              href="/blog/7-of-the-best-examples-of-beautiful-blog-design"
            >
              7 of the Best Examples of Beautiful Blog Design
            </a>
          </h3>
          <time
            class="css-1muxnm4-ArticleTeaser"
            data-testid="article-teaser-label"
          >
            Jul 25, 2020
          </time>
          <div
            class="css-1muxnm4-ArticleTeaser"
            data-testid="article-teaser-summary"
          >
            Harum et neque voluptate voluptatem aut quis culpa modi. Enim similique provident et libero quod magni soluta. Repudiandae deserunt architecto itaque repellat perfe.
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
